modules/contact/src/constraints/ComputeFrictionalForceLMMechanicalContact.C

```diff
@@ -107,20 +113,59 @@ ComputeFrictionalForceLMMechanicalContact::residualSetup()
 }
 
 void
-ComputeFrictionalForceLMMechanicalContact::jacobianSetup()
+ComputeFrictionalForceLMMechanicalContact::communicateVelocities()
 {
-  residualSetup();
+#ifdef MOOSE_SPARSE_AD
```

Comment:
Would it make sense to make this class to require sparse patterns?

Reply:
Just wondering if we are going to be moving forward with sparse, global framework for mortar.

Reply:
yes we emit the run-time errors if someone tries to use these mortar objects without the global indexing configuration, but this is required so that we can even compile this code when not using global indexing